jaraco/collections/__init__.py

```diff
@@ -8,10 +8,20 @@
 import random
 import re
 from collections.abc import Container, Iterable, Mapping
-from typing import Any, Callable, Union
+from typing import TYPE_CHECKING, Any, Callable, Dict, TypeVar, Union, overload
 
 import jaraco.text
 
+if TYPE_CHECKING:
+    from _operator import _SupportsComparison
```

Comment:
A type-checking-only private name from a private module. I get the feeling you're not gonna be a fan of that import, but it is a *very useful* `TypeAlias` that joins multiple `Protocol`s together.

If you prefer we can request to move this and its protocols into the `_typeshed` fake module like we did in https://github.com/python/typeshed/pull/11890

Reply:
I don't have enough experience with advanced typing to assess the viability of this approach. I'm happy to roll with it for now.